src/nucleus/tt.ml

```diff
@@ -135,10 +133,12 @@ and instantiate es ?(lvl=0) ({term=e';assumptions;loc;} as e) =
   in
   match e' with
 
-    | Type -> {e with assumptions}
+    | Type -> {e with assumptions} (* XXX why do we need assumptions here? Type is well formed by itself *)
```

Comment:
I think what's confusing here is that `B` gets an assumpton but the proper place for the assumption is *the judgement itself*. I would never want to be in a situation where `Type` has assumptions hanging off it.